src/uv-common.c

```diff
@@ -91,7 +93,10 @@ void* uv__calloc(size_t count, size_t size) {
 }
 
 void* uv__realloc(void* ptr, size_t size) {
-  return uv__allocator.local_realloc(ptr, size);
+  if (size > 0)
+    return uv__allocator.local_realloc(ptr, size);
+  uv__free(ptr);
```

Comment:
Why explicitly freeing the memory here?

Reply:
That's up to the caller, it should distinguish between `realloc(p, n)` with `n > 0` and `n == 0`. We seem to be handling that correctly everywhere by virtue of only using `realloc()` to grow allocations, not shrink or free them.

(One exception in src/win/core.c but it handles the `n == 0` case.)